app/feature/feature-travel-certificate/src/main/kotlin/com/hedvig/android/feature/travelcertificate/ui/TravelCertificateHistory.kt

```diff
@@ -0,0 +1,422 @@
+        onContinue = onContinue,
+        navigateUp = navigateUp,
+        historyList = uiState.certificateHistoryList,
+      )
+    }
+  }
+}
+
+@Composable
+private fun ShowHistory(
+  onIconClick: () -> Unit,
+  onCertificateClick: (String) -> Unit,
+  onContinue: () -> Unit,
+  navigateUp: () -> Unit,
+  historyList: List<TravelCertificate>,
+) {
+  HedvigScaffold(
+    navigateUp = navigateUp,
+    modifier = Modifier.clearFocusOnTap(),
+    topAppBarText = stringResource(id = R.string.PROFILE_ROW_TRAVEL_CERTIFICATE),
+    topAppBarActions = {
+      IconButton(
+        onClick = { onIconClick() },
+        modifier = Modifier.size(40.dp),
+      ) {
+        Icon(
+          imageVector = Icons.Hedvig.Info,
+          contentDescription = stringResource(R.string.REFERRALS_INFO_BUTTON_CONTENT_DESCRIPTION),
+          modifier = Modifier.size(24.dp),
+        )
+      }
+    },
+  ) {
+    if (historyList.isEmpty()) {
+      Spacer(modifier = Modifier.weight(1f))
+      ShowInitialInfo()
+    } else {
+      ShowNotEmptyList(
+        list = historyList,
+        onCertificateClick = onCertificateClick,
+      )
+    }
+    Spacer(modifier = Modifier.weight(1f))
+    VectorInfoCard(
+      text = String.format(stringResource(id = R.string.travel_certificate_start_date_info), 45),
+      modifier = Modifier
+        .fillMaxWidth()
+        .padding(16.dp, 16.dp, 16.dp, 8.dp),
+    )
+    HedvigSecondaryContainedButton(
+      text = stringResource(R.string.travel_certificate_get_travel_certificate_button),
+      onClick = onContinue,
+      modifier = Modifier.padding(16.dp, 8.dp, 16.dp, 0.dp),
+    )
+    Spacer(Modifier.height(32.dp))
+  }
+}
+
+@Composable
+private fun ShowInitialInfo() {
+  Column(
+    modifier = Modifier.fillMaxSize(),
+    horizontalAlignment = Alignment.CenterHorizontally,
+    verticalArrangement = Arrangement.Center,
+  ) {
+    HedvigInformationSection(
+      stringResource(id = R.string.travel_certificate_empty_list_message),
+    )
+  }
+}
+
+@Composable
+private fun ShowNotEmptyList(list: List<TravelCertificate>, onCertificateClick: (String) -> Unit) {
+  Column(
+    horizontalAlignment = Alignment.CenterHorizontally,
+  ) {
+    for (i in list) {
+      CertificateCard(
+        certificate = i,
+        onCertificateClick = onCertificateClick,
+      )
+    }
+  }
+}
+
+@Composable
+private fun CertificateCard(
+  certificate: TravelCertificate,
+  onCertificateClick: (String) -> Unit,
+  modifier: Modifier = Modifier,
+) {
+  Row(
+    verticalAlignment = Alignment.CenterVertically,
+    modifier = modifier
+      .fillMaxWidth()
+      .clickable(onClick = { onCertificateClick(certificate.signedUrl) })
+      .padding(16.dp),
+  ) {
+    val isExpired = isExpired(certificate.expiryDate)
+    val color = if (isExpired) Color.Red else Color.Unspecified
+    val expiryDateText =
+      "${certificate.expiryDate.dayOfMonth} ${certificate.expiryDate.month.name.lowercase().substring(0, 3)}"
```

Comment:
Could we look into using this https://github.com/HedvigInsurance/android/blob/0baeb902323e5e6443bdf61aee2f1ae4e59e2147/app/core/core-ui/src/main/kotlin/com/hedvig/android/core/ui/HedvigDateTimeFormatter.kt#L65-L67 perhaps here to do the formatting? It's nice to try and be consistent in the codebase in how we do this.

Try to use that API and if it's hard to use LMK. Inside a composable you can get the locale with `getLocale()` here https://github.com/HedvigInsurance/android/blob/7e85f0db5781027e32700587c32c517454c95dac/app/core/core-ui/src/main/kotlin/com/hedvig/android/core/ui/GetLocale.kt#L12